### ISSUE TITLE
Some HTML input element props don't get set using IHtmlInputElement interface

### ISSUE THREAD

**Issue by j055** [AngleSharp/AngleSharp#471]
I noticed that some attributes don't get set when using the IHtmlInputElement interface:

```
    [TestClass]
    public class AngleSharpIHtmlInputElementTests
    {
        private static IHtmlDocument CreateDocument()
        {
            var htmlParser = new AngleSharp.Parser.Html.HtmlParser();
            var document = htmlParser.Parse(string.Empty);
            return document;
        }

        [TestMethod]
        public void SetTextTypeInputAttributes()
        {
            var document = CreateDocument();

            var input = document.CreateElement<IHtmlInputElement>();
            input.Name = "name";
            input.Type = "text";
            input.MaxLength = 500;
            input.Value = "value";
            input.IsRequired = true;

            document.Body.AppendElement(input);

            Assert.IsTrue(document.QuerySelector("input").HasAttribute("name"), "name attribute not set");
            Assert.IsTrue(document.QuerySelector("input").HasAttribute("type"), "type attribute not set");
            Assert.IsTrue(document.QuerySelector("input").HasAttribute("maxlength"), "maxlength attribute not set");
            Assert.IsTrue(document.QuerySelector("input").HasAttribute("required"), "required attribute not set");
            Assert.IsTrue(document.QuerySelector("input").HasAttribute("value"), "value attribute not set");
        }


        [TestMethod]
        public void SetCheckboxTypeInputAttributes()
        {
            var document = CreateDocument();

            var input = document.CreateElement<IHtmlInputElement>();
            input.Name = "name";
            input.Type = "checkbox";
            input.Value = "value";
            input.IsRequired = true;
            input.IsChecked = true;

            document.Body.AppendElement(input);

            Assert.IsTrue(document.QuerySelector("input").HasAttribute("name"), "name attribute not set");
            Assert.IsTrue(document.QuerySelector("input").HasAttribute("type"), "type attribute not set");
            Assert.IsTrue(document.QuerySelector("input").HasAttribute("maxlength"), "maxlength attribute not set");
            Assert.IsTrue(document.QuerySelector("input").HasAttribute("required"), "required attribute not set");
            Assert.IsTrue(document.QuerySelector("input").HasAttribute("checked"), "checked attribute not set");
            Assert.IsTrue(document.QuerySelector("input").HasAttribute("value"), "value attribute not set");
        }
    }
```

**Comment by FlorianRappl** [comment]
Please check with your browser first. In general: **properties** are not **attributes**. It is right, that *many* properties refer to an attribute (some expose the attribute directly, others filtered), however, not *all* properties do that.

The `value` attribute of an `input` element is only used to define the `default` (i.e., initial) value. The *live* value is an *internal state*. This state is not exposed as an attribute!

Your second test is flawed anyway. You are not manipulating the `maxlength` attribute - so don't check for it! Besides this, the `checked` attribute behaves similar to the `value` attribute. Since the property is a boolean and attributes are always strings you should have been more skeptical.

Long story short: Before reporting a bug here, please do at least minimal testing in *any* browser. Thanks anyway for your efforts!
